### DISCUSSION TITLE
Ideas for KerasCV tutorials

### DISCUSSION THREAD

**Discussion by jaygala223** [keras-team/keras-cv#1483] [Ideas] [answered]
Hello, I am interested in contributing tutorials for KerasCV as part of GSoC 2023 and was hoping the maintainers could look at my ideas below and give me some feedback.

## Beginner
- [ ] Introduction to computer vision and deep learning with KerasCV
- [ ] Installing KerasCV and its dependencies
- [ ] Understanding the KerasCV API and its key components
- [ ] Basic image operations such as loading, resizing, and saving images
- [ ] Building simple image classification models using pre-trained models and custom models
- [ ] Introduction to image augmentation techniques such as flipping, rotating, and cropping
- [ ] Training and evaluating image classification models including model selection and hyperparameter tuning
- [ ] Fine-tuning pre-trained models for custom image classification tasks

## Intermediate
- [ ] Introduction to object detection and localization including bounding box regression and non-maximum suppression
- [ ] Building object detection models using pre-trained models and custom models
- [ ] Fine-tuning pre-trained object detection models for custom datasets
- [ ] Introduction to image segmentation and instance segmentation
- [ ] Building image segmentation models using pre-trained models and custom models
- [ ] Fine-tuning pre-trained image segmentation models in KerasCV for custom datasets
- [ ] Introduction to transfer learning including feature extraction and fine-tuning techniques
- [ ] Advanced image augmentation techniques in KerasCV, such as CutMix, MixUp, and RandAugment

## Advanced
- [ ] Introduction to generative models with KerasCV, including GANs and VAEs
- [ ] Building generative models using pre-trained models and custom models
- [ ] Fine-tuning pre-trained generative models for custom datasets
- [ ] Introduction to style transfer and neural style transfer with KerasCV
- [ ] Building style transfer models using pre-trained models and custom models

**Comment by jaygala223** [comment]
Tagging @LukeWood and @jbischof for review

**Comment by jbischof** [comment, marked as answer]
Apologies @jaygala223 GSoC has its own process/bureaucracy and is now closed for the year. Let us know if you would still like to contribute outside the problem. We definitely need to return to segmentation soon.

**Reply by jaygala223** [reply]
Yes, I understand. I would love to contribute outside this as well. How can I help?

**Reply by jbischof** [reply]
Awesome! @ianstenbit is cleaning up our issues and will find a good starter task

**Reply by ianstenbit** [reply]
Here's an issue that could be a nice thing to start with:
https://github.com/keras-team/keras-cv/issues/1784

I left a comment there about some particular patterns to look out for that could be causing these issues.

Please feel free to @ me if you have questions 😃

**Reply by jaygala223** [reply]
Thanks for sharing @ianstenbit and @jbischof. I'm on it 👍
